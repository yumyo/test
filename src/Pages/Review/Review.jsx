import React from 'react';
import Basket from './components/Basket/Basket.jsx';
import TicketList from './components/TicketList/TicketList.jsx';
import Total from './components/Total/Total.jsx';
import './Review.scss';

const Review = ({ Campaign = {} }) => {
    const { Products, title } = Campaign;
    return (
        <div className="flex-layout">
            {[
                (
                    <TicketList
                        key="listings"
                        title={title}
                        products={Products || []}
                    />
                ),
                (
                    <div className="basket">
                        <Basket />
                        <Total />
                    </div>
                ),
            ]}
        </div>
    );
}

export default Review;
