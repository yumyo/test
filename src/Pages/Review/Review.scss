.flex-layout {
  display: flex;
  flex-direction: row;
}
