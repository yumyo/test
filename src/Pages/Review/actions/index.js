
export const addToBasket = (action) => Object({
    type: 'ADD_TO_BASKET',
    id: action.id,
    title: action.title,
    price: action.price,
});


export const removeFromBasket = (action) => Object({
    type: 'REMOVE_FROM_BASKET',
    id: action.id,
    title: action.title,
    price: action.price,
});
