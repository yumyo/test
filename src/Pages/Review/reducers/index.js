import { combineReducers } from 'redux'
import basket from './basket.js'

const reviewApp = combineReducers({
  basket,
});

export default reviewApp;
