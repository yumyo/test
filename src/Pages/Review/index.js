import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import withQuery from './components/Query/Query.jsx';
import Review from './Review.jsx';
import state from './reducers';

const store = createStore(state);
const WrpCmp = withQuery(Review);

const ReviewIndex = (props) => (
  <Provider store={store}>
    <WrpCmp {...props} />
  </Provider>
);

export default ReviewIndex;
