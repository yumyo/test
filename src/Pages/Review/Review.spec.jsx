import React from 'react';
import { shallow } from 'enzyme';
import Review from './Review.jsx';

describe('<Review />', () => {
    let component;
    const props = {
        Campaign: {
            title: 'test campaign',
            Products: [
                {
                    id: '001',
                    title: 'xxx',
                    price: 40
                }
            ]
        },
    }

    beforeEach(() => {
        component = shallow(<Review {...props} />);
    });

    it('renders with props', () => {
        expect(component).toMatchSnapshot();
    });
});
