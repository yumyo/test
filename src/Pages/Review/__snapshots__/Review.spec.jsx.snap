// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Review /> renders with props 1`] = `
<div
  className="flex-layout"
>
  <TicketList
    key="listings"
    products={
      Array [
        Object {
          "id": "001",
          "price": 40,
          "title": "xxx",
        },
      ]
    }
    title="test campaign"
  />
  <div
    className="basket"
  >
    <Connect(BasketComponent) />
    <Connect(TotalComponent) />
  </div>
</div>
`;
